arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/ForwardCancellable.kt

```diff
@@ -18,35 +18,37 @@ import kotlin.coroutines.suspendCoroutine
  *
  *  - When `cancel` is called after `complete`, then `cancel` immediately runs the cancellation token passed to `complete`.
  *
- *  - When the `ForwardCancellable` is completed, before `cancel`'s token is invoked.
- *  The completed [CancelToken] will be invoked immediately when `cancel`'s token is invoked.
- *
  *  - Calling `complete` twice will result in an [ArrowInternalException], and is considered an internal developer bug.
  *
  * You can find example of this usage in [bracket] and [guaranteeCase] to schedule the `cancel` frame and
  * in [uncancellable] to back-pressure the original context until the uncancellable context is finished.
+ *
+ * @param ctx, the CoroutineContext where the `CancelToken` is run on.
+ *   => This is taken into the constructor because we want to run it on the `CoroutineContext` of the creator of the `CancelToken`,
+ *   and not on the `CoroutineContext` of the caller of `cancel`.
+ *   => We want the `CancelToken` from `cancellable` to run on the `CoroutineContext` where `cancellable` is  called.
+ *   => We want the release function of `bracket` & `bracketCase` to run on the `CoroutineContext` where `bracketX` is called.
  */
-internal class ForwardCancellable {
+internal class ForwardCancellable(private val ctx: CoroutineContext) {
```

Comment:
Yes, this could definitely be configurable. I'm still wondering a bit, since running cancel tokens on an `IOPool` is probably safest.

Now the're most likely to run on the default pool, but for the current usages I think this makes most sense. It's used in `bracketCase` and it's derived methods, and `cancellable`. It makes the `release` and `CancelToken` to be run with the same `ctx` as their `use/acquire` and `f` functions.